x/crosschain/types/keys.go

```diff
@@ -75,6 +76,10 @@ func (m CrossChainTx) LogIdentifierForCCTX() string {
 
 }
 
+func FinalizedInboundKey(intxHash string, chainID int64, eventIndex uint64) string {
+	return fmt.Sprintf("%s-%d-%d", intxHash, chainID, eventIndex)
```

Comment:
Would it be a good idea to reorder the key as `chainID-intxHash-eventIndex`. In future we may have a query to easily collect all finalized inTxs by chain and to know which ones are processed which ones are not.

Reply:
Agree on this!